lib/hawkular/base_client.rb

```diff
@@ -127,6 +127,25 @@ def generate_query_params(params = {})
       end
     end
 
+    # Generate a new url with the passed sufix path if the path is not already added
+    # to the URL
+    # @param entrypoint [String] base path
+    # @param sufix_path [String] sufix path to be added if it doesn't exist
+    # @return [String] URL with path attached to it at the end
+    def normalize_entrypoint_url(entrypoint, sufix_path)
+      relative_path_rgx = Regexp.new("#{Regexp.quote(sufix_path)}(\/)*$")
```

Comment:
I also recommend to test: IP instead of hostname, without port (when it's 80 it's typically omitted)